src/server/qgsserversettings.h

```diff
@@ -294,6 +296,27 @@ class SERVER_EXPORT QgsServerSettings
      */
     QString serviceUrl( const QString &service ) const;
 
+    /**
+     * Returns the config cach check interval for the 'interval' strategy.
+     * \since QGIS 3.24
+     */
+    int projectCacheCheckInterval() const;
+
+    /**
+     * Returns the project's cache strategy
+     * The default value is 'filesystem', the value can be change by setting the environment
+     * variable QGIS_SERVER_PROJECT_CACHE_STRATEGY.
+     * Possible values are:
+     *
+     * - 'fileystem': Use file system watcher for notifying projects change. Note that it works
+     *   only with projects stored in files and not across mounted NFS volumes on Linux.
+     * - 'periodic': Timer based periodic check for project's changes. Works with all storage backend.
+     * - 'off': Disable completely internal project's cache handling
+     *
+     * \since QGIS 3.24
+     */
+    QString projectCacheStrategy() const;
```

Comment:
What do you think about defining an enum for possible values instead of plain strings?

Reply:
Not sure that it will bring improvement since stragegy classes are internally  instancied directly  from settings values.

Reply:
I think that string comparisons should be avoided when possible. It's not robust and hardly readable compared to enum. BTW QGIS Server 2.X source code was very hard to maintain and buggy notably for this reason.

I won't block this PR (especially if @elpaso and @m-kuhn are ok with the current implementation) but I'll probably take care of it at one time or another.

Reply:
> I think that string comparisons should be avoided when possible

Whatever you do you can prevent string comparison when converting the settings to enum and there is nowhere else than initialisation where such comparison is done.

Reply:
it always start this way